src/services/edusharing/logic/connector.js

```diff
@@ -0,0 +1,277 @@
+			if (result) {
+				const parsedResult = JSON.parse(result);
+				return parsedResult.access_token;
+			}
+			// eslint-disable-next-line no-console
+			console.error('Oauth failed');
+			return null;
+		});
+	}
+
+	allConfigurationValuesHaveBeenDefined() {
+		return (
+			Configuration.get('ES_DOMAIN')
+			&& Configuration.get('ES_USER')
+			&& Configuration.get('ES_PASSWORD')
+			&& Configuration.get('ES_GRANT_TYPE')
+			&& Configuration.get('ES_OAUTH_SECRET')
+			&& Configuration.get('ES_CLIENT_ID')
+		);
+	}
+
+	async requestRepeater(options) {
+		let retry = 0;
+		const errors = [];
+		do {
+			try {
+				const eduResponse = await request(options);
+				return JSON.parse(eduResponse);
+			} catch (e) {
+				// eslint-disable-next-line no-console
+				if (e.statusCode === 401) {
+					logger.info(`Trying to renew Edu Sharing connection. Attempt ${retry}`);
+					await this.login();
+				} else {
+					logger.error(e);
+					errors.push(e);
+				}
+			}
+			retry += 1;
+		} while (retry < 3);
+
+		throw new GeneralError('Edu Sharing Retry failed', errors);
+	}
+
+	configurationIncompleteError() {
+		return new Forbidden(
+			'Not all ES_... vars have been defined, update configuration.',
+		);
+	}
+
+
+	async login() {
+		this.authorization = await this.getCookie();
+		this.accessToken = await this.getAuth();
+	}
+
+	isLoggedin() {
+		// returns false if cookie or accesstoken is falsy
+		return !!this.authorization && !!this.accessToken;
+	}
+
+	async GET(id, params) {
+		if (!this.allConfigurationValuesHaveBeenDefined()) {
+			throw this.configurationIncompleteError();
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const propertyFilter = '-all-';
+
+		const options = {
+			method: 'GET',
+			// eslint-disable-next-line max-len
+			url: `${Configuration.get('ES_DOMAIN')
+			}/edu-sharing/rest/node/v1/nodes/mv-repo.schul-cloud.org/${id
+			}/metadata?propertyFilter=${propertyFilter}`,
+			headers: {
+				...EduSharingConnector.headers,
+				cookie: this.authorization,
+			},
+			timeout: REQUEST_TIMEOUT,
+		};
+
+		const eduResponse = await this.requestRepeater(options);
+		return eduResponse.node;
+	}
+
+	async FIND(data) {
+		const contentType = data.query.contentType || 'FILES'; // enum:[FILES,FILES_AND_FOLDERS,COLLECTIONS,ALL]
+		const skipCount = parseInt(data.query.$skip, 10) || 0;
+		const maxItems = parseInt(data.query.$limit, 10) || 9;
+		const sortProperties = data.query.sortProperties || 'score';
+		const sortAscending = false;
+		const propertyFilter = '-all-'; // '-all-' for all properties OR ccm-stuff
+		const searchWord = data.query.searchQuery || '';
+
+		if (!this.allConfigurationValuesHaveBeenDefined()) {
+			throw this.configurationIncompleteError();
+		}
+
+		if (this.isLoggedin() === false) {
+			await this.login();
+		}
+
+		const urlBase = `${Configuration.get('ES_DOMAIN')
+		}/edu-sharing/rest/search/v1/queriesV2/mv-repo.schul-cloud.org/mds/ngsearch/?`;
+		const url = urlBase
```

Comment:
create query: cerate object containing all these properties (contentType, skipCount, ...) and use url lib or something already in use